build/Dockerfile

```diff
@@ -0,0 +1,21 @@
+
+FROM golang:1.18.6-alpine3.15 AS builder
```

Comment:
We'll need to change from alpine to debian as we are moving away from alpine. Go doesn't support muslc properly and we have run into issues due to it. I'd suggest `1.18.6-buster` and then `debian:stable` perhaps for the runner image?

Any suggestions here @phoinixgrr?

Reply:
> We'll need to change from alpine to debian as we are moving away from alpine. Go doesn't support muslc properly and we have run into issues due to it. I'd suggest `1.18.6-buster` and then `debian:stable` perhaps for the runner image?
> 
> Any suggestions here @phoinixgrr?
> 
> 

Fully ok with what is proposed here.

Some context on the decision making for mattermost server: https://github.com/mattermost/mattermost-server/pull/20807

Reply:
I would propose a multistage build with debian as builder and 'distroless' as runner.

Reply:
Sounds good. Did you have any specific distroless image in mind? I see there are a couple of options.

Reply:
gcr.io/distroless/static should work fine

Reply:
Sorry guys, but I was not able to make it work. I will leave it as is.

Reply:
Could you paste the errors that you faced? This is still in my personal repo, so we can merge it right now, and work on the improvements later.